google/cloud/pubsub/internal/tracing_helpers.cc

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
+#include "google/cloud/pubsub/internal/tracing_helpers.h"
```

Comment:
I wonder whether this inclusion should be unconditional.

Reply:
Carlos had the same thought. Done.

Reply:
It looks like the inclusion of the header is still conditional.  But even more, I now wonder about the new tactic to have the header declare the function unconditionally, but the implementation to define it conditionally.  It seems like they should match.

Reply:
Maybe it should all be inside the ifdefs. I'm not really sure the argument against it. It has otel symbols in it either way and should not be included if there is no otel.